### DISCUSSION TITLE
multiple vercel json for nx integrated monorepo approach

### DISCUSSION THREAD

**Discussion by saibs** [vercel/community#1234] [Help] [answered]
### Summary

Hi,
We are using NX [integrated monorepo ](https://nx.dev/concepts/integrated-vs-package-based) and have multiple react SPA. 
As for each app, we are able to run the following from root app
`npx nx run app:build --prod` and give an output folder path.
`dist/apps/app`
with empty root directory. Is it possible to configure the path to vercel json file for each app?

Unlike package based monorepo we can have multiple vercel json files as each project has different root directory.

### Additional information

- We have tried moving vercel json from the app folder to root of monorepo during pre installation script and it didn't work.
- The only thing that works is having the right vercel json already at the root of the monorepo. But this is not ideal as we have a combination of SPA, SSG apps and only SPA apps needs rewrite to index page.

**Comment by saibs** [comment, marked as answer]
So this finally worked vercel json in each app folder and then the following 
build command
`cd ../.. && nx run app:build --configuration=production --outputPath=apps/app/dist`
output directory
`dist`
root directory
`apps/app`

**Reply by mathura333** [reply]
Is there any other workaround as well?

I actually want my builds to be in the dist/apps/{appName} only

**Reply by mathura333** [reply]
I tried moving the vercel.json before build to root of repo but that also did not work

**Reply by glowdragon** [reply]
@mathura333 Change the output directory to `../../dist/apps/app`

**Comment by aiyeola** [comment]
to buttress the reply to this discussion

https://github.com/orgs/vercel/discussions/2074#discussioncomment-8364957
